text/353-cfn-registry-constructs.md

```diff
@@ -0,0 +1,467 @@
+includes a construct class and all the relevant data types for this
+type.
+
+The module version corresponds to the version of the type schema
+in the public registry.
+
+---
+
+Ticking the box below indicates that the public API of this RFC has been
+signed-off by the API bar raiser (the `api-approved` label was applied to the
+RFC pull request):
+
+```
+[ ] Signed-off by API Bar Raiser @xxxxx
+```
+
+## Public FAQ
+
+### What are we launching today?
+
+We are publishing a set of construct libraries (in all JSII languages) which include constructs (and auxiliary types) for 
+all the resources and modules in the public CloudFormation registry.
+
+### Why should I use this feature?
+
+This library makes it easier to discover and use CloudFormation extensions in CDK apps.
+
+Previously, in order to use a resource from the CloudFormation registry, you would need to look up the resource documentation
+and use the low-level, weakly-typed `CfnResource` class to define it in your CDK app:
+
+```ts
+new CfnResource(this, 'MyEksCluster', {
+  type: 'AWSQS::EKS::Cluster',
+
+  // weakly-typed!
+  properties: {
+    name: 'my-new-cluster',
+    tags: [ { key: 'foo', value: 'bar' } ],
+  },
+});
+```
+
+With `@cdk-cloudformation-extensions` all public resources and modules will be listed in the **Construct Hub** like any
+other construct and by importing the relevant `cdk-cloudformation-extensions` module into their projects, they will be able to
+use them via strongly-typed classes. IDEs will show type information and inline help derived from the
+extension schema.
+
+```ts
+import { Cluster } from '@cdk-cloudformation-extensions/awsqs-eks';
+
+new Cluster(this, 'MyEksCluster', {
+  name: 'my-new-cluster',
+  tags: [ { key: 'foo', value: 'bar' } ],
+});
+```
+
+## Internal FAQ
+
+### Why are we doing this?
+
+We are doing this because CDK users would like to be able to use CloudFormation extensions 
+(modules, resources) as first-class citizens in CDK code. Extensions and native resource types
+should look and feel the same way when defined in CDK apps.
+
+Additionally, we would like to surface CloudFormation extensions in the upcoming Construct Hub
+and to that end, if we simply publish a construct library that includes constructs for all the public,
+extensions, this library will naturally be discoverable through the Hub.
+
+### Why should we _not_ do this?
+
+We have a longer term plan ([RFC#77](https://github.com/aws/aws-cdk-rfcs/issues/77)) to add 
+support for a CLI command called `cdk import` which will allow users to generate L1s from any
+registry type just-in-type and add them to their project (this is similar to CDK8s and CDKtf).
+
+### What is the technical solution (design) of this feature?
+
+The idea is to do something similar to what we do with the CloudFormation L1s 
+in the AWS CDK and simply publish JSII construct libraries which includes 
+statically-generated L1s for all the public registry extensions.
+
+#### Library per type
```

Comment:
I like the simplicity of this solution, however we should be careful.

To get a sense of scale, how many public types are there today, in orders of magnitude?

Are there limits in the target pkg managers that we will hit here?

It could be useful to publish these under a separate account in these pkg managers to keep the blast radius contained if we hit a hard limit.
I actually don't know how we would wiggle out if we hit such a limit and aren't able to publish any more modules. Might be worth giving some more thought?

Reply:
> To get a sense of scale, how many public types are there today, in orders of magnitude?

There are currently 44 public types (resources & modules), but we expect this list to grow.

> Are there limits in the target pkg managers that we will hit here?

Doesn't seem like there are any constraints on the number of packages in a scope in Npm. But if we will hit any limits, we can decide to split into multiple scopes/accounts/whatever. Since these modules will be listed in the construct hub their discoverability is not predicated on the scope/group-id or anything.

Reply:
Added a note.